Samples/ParlaMint-IL/2021/ParlaMint-IL_2021-12-21.xml

```diff
@@ -0,0 +1,243 @@
+            <bibl>
+               <title type="main" xml:lang="he">פרוטוקולים של הכנסת</title>
+               <title type="main" xml:lang="en">Knesset Protocols</title>
+               <idno type="URI" subtype="parliament">https://www.knesset.gov.il</idno>
+               <date from="1993-07-12" to="2024-04-03">1993-07-12 - 2024-04-03</date>
+            </bibl>
+         </sourceDesc>
+      </fileDesc>
+      <encodingDesc>
+         <projectDesc>
+            <p xml:lang="en">
+               <ref target="https://www.clarin.eu/parlamint">ParlaMint</ref> is a
+    project that aims to (1) create a multilingual set of comparable corpora of parliamentary
+    proceedings uniformly encoded according to the
+    <ref target="https://clarin-eric.github.io/ParlaMint/">ParlaMint encoding guidelines</ref>, 
+    covering the period from 2015 to mid-2022; (2) add linguistic annotations to the corpora and
+    machine-translate them to English; (3) make the corpora available through concordancers; and
+    (4) build use cases in Political Sciences and Digital Humanities based on the corpus
+    data.</p>
+            <p xml:lang="he">הקורפוס הוא חלק מפרויקט ParlaMint, שמטרתו לספק סט רב לשוני של קורפוסים פרלמנטריים.</p>
+         </projectDesc>
+         <tagsDecl>
+            <namespace name="http://www.tei-c.org/ns/1.0">
+               <tagUsage gi="body" occurs="1"/>
+               <tagUsage gi="div" occurs="1"/>
+               <tagUsage gi="head" occurs="1"/>
+               <tagUsage gi="seg" occurs="99"/>
+               <tagUsage gi="text" occurs="1"/>
+               <tagUsage gi="u" occurs="11"/>
+            </namespace>
+         </tagsDecl>
+      </encodingDesc>
+      <profileDesc>
+         <settingDesc>
+            <setting>
+               <name>Knesset 24</name>
+               <date when="2021-12-21"/>
+            </setting>
+         </settingDesc>
+      </profileDesc>
+      <revisionDesc>
+         <change when="2024-11-20">Initial conversion to TEI format.</change>
+      </revisionDesc>
+   </teiHeader>
+   <text ana="#covid">
+      <body>
+         <div type="debateSection" xml:id="session.24_ptv_616837_doc">
+            <head xml:id="ParlaMint-IL_2021-12-21.head1">ועדת החינוך, התרבות והספורט</head>
+            <u xml:id="u.session.24_ptv_616837_doc.0"
+               who="#person.30118"
+               ana="#chair">
+               <seg xml:id="seg.id-317051d3-0083-40c6-babc-4f111f66e60d">בוקר טוב לכולם, בוקר טוב לכל הנוכחים.</seg>
+               <seg xml:id="seg.id-8005043c-61d6-48d2-9674-c5ff64c438e6">אני פותחת את ישיבת ועדת החינוך, התרבות והספורט של הכנסת.</seg>
+               <seg xml:id="seg.id-997bd044-dd0c-46d3-aaf4-cca2871481d6">קודם כל, אנחנו נפתח בהקמות שתי ועדות משנה.</seg>
+               <seg xml:id="seg.id-145ec7a4-00e8-4797-a646-e34a01239a91">הקמת ועדת משנה ראשונה היא לחוק חינוך ממלכתי דתי.</seg>
+               <seg xml:id="seg.ec44704c-bf27-4135-85ef-90a2ec33938e">אבקש להביא לידיעת</seg>
```

Comment:
`<seg>` element corresponds to paragraphs in source transcription: https://clarin-eric.github.io/ParlaMint/#sec-uterrance
> The utterances are then segmented using the [<seg>](https://clarin-eric.github.io/ParlaMint/#TEI.seg) element, which encodes the paragraphs of the source transcription. Even if the source files do not contain paragraph markings, each speech should contain at least one segment.

You are using `seg` for some sub-sentence segmentation, which causes wrong linguistic annotations - because `אבקש להביא לידיעת` is a sentence-fragment (I believe).

This utterance should contain two paragraphs according to the source HTML: (https://oknesset.org/meetings/2/1/2166933.html#speech-2166933-7)

BTW, there are recommended different ID structures in documentation - it is better to use a structure like this: `ParlaMint-CZ_2020-01-22-ps2017-040-02-005-012.u1.p1`
it makes debugging more manageable, and you can be sure that the id is unique (I don't know how your hashes are constructed, but I guess the "good morning" sentence repeats in the corpus, so if it is only built from text content, then it is not unique)

Reply:
Are the texts segmented correctly? Is not this bug in the source Knesset Corpus?
Converting your corpus into a ParlaMint format is an excellent opportunity to discover source corpus bugs so that you can fix them in the source corpus(in the next release) and also in the ParlaMint version.

From ParlaMint's point of view (my opinion), it makes better sense to create one paragraph(`seg`) per utterance(`u`) if you decide to ignore paragraphs in original transcriptions (cause you are using an existing corpus which does not care about paragraphs).
But considering right-left Hebrew writing, I am not a hundred per cent sure that it is a good idea because I don't see all the consequences that can arise...

Reply:
Thank you for your feedback and suggestions. I have merged all the sentences back into one paragraph per utterance, as recommended. 
Additionally, I have fixed the ud logic in my code that caused some of the issues, as well as improved the fallback logic.
Regarding the segmentation, you are correct that there are some bugs, as the process was done automatically. However, the error rate is relatively low. Since the data is already segmented into sentences, addressing these issues at this stage would be quite challenging. I do plan to address this in the future as part of ongoing improvements.
Please let me know if anything else is needed for the data to meet the ParlaMint standards or to proceed with merging this pull request.

Reply:
> Please let me know if anything else is needed for the data to meet the ParlaMint standards or to proceed with merging this pull request.

Tonight/tomorrow, I will give you more complex feedback, including reporting bugs and suggestions for improvement to make your corpus more comparable with other ParlaMint corpora.